src/app/InteractionModelEngine.cpp

```diff
@@ -280,6 +283,59 @@ CHIP_ERROR InteractionModelEngine::OnInvokeCommandRequest(Messaging::ExchangeCon
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR InteractionModelEngine::ParseAttributePaths(Access::SubjectDescriptor & subjectDescriptor,
+                                                       AttributePathIBs::Parser & attributePathListParser, bool & hasValidPath,
+                                                       size_t & requestedAttributePathCount)
+{
+    TLV::TLVReader pathReader;
+    attributePathListParser.GetReader(&pathReader);
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    hasValidPath                = false;
+    requestedAttributePathCount = 0;
+
+    while (CHIP_NO_ERROR == (err = pathReader.Next()))
+    {
+        VerifyOrReturnError(TLV::AnonymousTag() == pathReader.GetTag(), CHIP_ERROR_INVALID_TLV_TAG);
+
+        AttributePathIB::Parser path;
+        ReturnErrorOnFailure(path.Init(pathReader));
+
+        AttributePathParams params;
+        ReturnErrorOnFailure(path.ParsePath(params));
+
+        //
+        // We create an iterator to point to a single item object list that tracks the path we just parsed.
+        // This avoids the 'parse all paths' approach that is employed in ReadHandler since we want to
+        // avoid allocating out of the path store during this minimal initial processing stage.
+        //
+        ObjectList<AttributePathParams> paramsList;
+        paramsList.mValue = params;
+        AttributePathExpandIterator pathIterator(&paramsList);
+        ConcreteAttributePath readPath;
+
+        for (; pathIterator.Get(readPath); pathIterator.Next())
+        {
+            Access::RequestPath requestPath{ .cluster = readPath.mClusterId, .endpoint = readPath.mEndpointId };
+            err = Access::GetAccessControl().Check(subjectDescriptor, requestPath, RequiredPrivilege::ForReadAttribute(readPath));
```

Comment:
Is this going to be accompanied by spec changes?  Because at least for reads the ACL checks come after various other checks in the spec right now (like whether the path exists at all).  It's probably fine to not expose that information if you would not have ACLs for it anyway, but needs some careful thought about consequences.

Reply:
I'm intentionally not doing ACL checks to minimize duplication of work with what ReadHandler does later anyways. I can leave a comment to that effect here.

Reply:
Spec PR is [out](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/5241), but I think it still needs some work....